public/contextMenu.js

```diff
@@ -0,0 +1,258 @@
+      cut: decorateMenuItem({
+        id: 'cut',
+        label: 'Cu&t',
+        enabled: can('Cut'),
+        visible: props.isEditable,
+        click(menuItem) {
+          const target = webContents(win);
+
+          if (!menuItem.transform && target) {
+            target.cut();
+          } else {
+            props.selectionText = menuItem.transform ? menuItem.transform(props.selectionText) : props.selectionText;
+            electron.clipboard.writeText(props.selectionText);
+          }
+        }
+      }),
+      copy: decorateMenuItem({
+        id: 'copy',
+        label: '&Copy',
+        enabled: can('Copy'),
+        visible: props.isEditable || hasText,
+        click(menuItem) {
+          const target = webContents(win);
+
+          if (!menuItem.transform && target) {
+            target.copy();
+          } else {
+            props.selectionText = menuItem.transform ? menuItem.transform(props.selectionText) : props.selectionText;
+            electron.clipboard.writeText(props.selectionText);
+          }
+        }
+      }),
+      paste: decorateMenuItem({
+        id: 'paste',
+        label: '&Paste',
+        enabled: editFlags.canPaste,
+        visible: props.isEditable,
+        click(menuItem) {
+          const target = webContents(win);
+
+          if (menuItem.transform) {
+            let clipboardContent = electron.clipboard.readText(props.selectionText);
+            clipboardContent = menuItem.transform ? menuItem.transform(clipboardContent) : clipboardContent;
+            target.insertText(clipboardContent);
+          } else {
+            target.paste();
+          }
+        }
+      }),
+      inspect: () => ({
+        id: 'inspect',
+        label: 'I&nspect Element',
+        click() {
+          win.inspectElement(props.x, props.y);
+
+          if (webContents(win).isDevToolsOpened()) {
+            webContents(win).devToolsWebContents.focus();
+          }
+        }
+      }),
+    };
+
+    const shouldShowInspectElement = typeof options.showInspectElement === 'boolean' ? options.showInspectElement : false;
+
+    let dictionarySuggestions = [];
+
+    let menuTemplate = [
+      defaultActions.cut(),
+      defaultActions.copy(),
+      defaultActions.paste(),
+      defaultActions.separator(),
+      shouldShowInspectElement && defaultActions.inspect(),
+      defaultActions.separator()
+    ];
+
+    if (options.menu) {
+      menuTemplate = options.menu(defaultActions, props, win, dictionarySuggestions, event);
+    }
+
+    if (options.prepend) {
+      const result = options.prepend(defaultActions, props, win, event);
+
+      if (Array.isArray(result)) {
+        menuTemplate.unshift(...result);
+      }
+    }
+
+    if (options.append) {
+      const result = options.append(defaultActions, props, win, event);
+
+      if (Array.isArray(result)) {
+        menuTemplate.push(...result);
+      }
+    }
+
+    menuTemplate = removeUnusedMenuItems(menuTemplate);
```

Comment:
What's this doing?

Reply:
The lib was structured in a way that it first includes all menuItems and then runs a `can` function to remove those that are not relevant for the runtime.

Since we won't handle a dynamic amount of menuItems we could do it the other way around, checking if the elements are needed before adding, but I don't think it will improve the code in a meaningful way.

I'll add some inline documentation to improve the code readability.